core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -980,7 +980,10 @@ public void run() {
         public void onNodeAck(DiscoveryNode node, @Nullable Throwable t) {
             if (!ackedTaskListener.mustAck(node)) {
                 //we always wait for the master ack anyway
-                if (!node.equals(nodes.getMasterNode())) {
+                DiscoveryNode masterNode = nodes.getMasterNode();
```

Comment:
previously, equals did only check for equality of ephemeral id. Extending this check now to all attributes felt:
1) inefficient (always compare all attributes just to detect that it's same node)
2) unsafe w.r.t. future extensions (where we could for example make it possible for node to have attributes changing at runtime).